File: src/main/java/io/junseok/mealmate/domain/restaurant/repository/QueryRestaurantRepository.java
package io.junseok.mealmate.domain.restaurant.repository;

import com.querydsl.core.types.dsl.BooleanExpression;
import com.querydsl.jpa.impl.JPAQueryFactory;
import io.junseok.mealmate.domain.restaurant.entity.QRestaurant;
import io.junseok.mealmate.domain.restaurant.entity.Restaurant;
import java.util.List;
import javax.persistence.EntityManager;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class QueryRestaurantRepository {
    private final EntityManager em;
    QRestaurant restaurant = QRestaurant.restaurant;
    public List<Restaurant> findAllRestaurant(String restaurantType){

        JPAQueryFactory queryFactory = new JPAQueryFactory(em);


        return queryFactory.selectFrom(restaurant)
            .where(restaurantTypeLike(restaurantType))
            .fetch();
    }

    private BooleanExpression restaurantTypeLike(String restaurantType){
        return !restaurantType.isEmpty() ? restaurant.restaurantType.eq(restaurantType):null;
    }
}

File: src/main/java/io/junseok/mealmate/domain/restaurant/repository/QueryRestaurantRepository.kt
package io.junseok.mealmate.domain.restaurant.repository

import com.querydsl.core.types.dsl.BooleanExpression
import com.querydsl.jpa.impl.JPAQueryFactory
import io.junseok.mealmate.domain.restaurant.entity.QRestaurant
import io.junseok.mealmate.domain.restaurant.entity.Restaurant
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Repository
import javax.persistence.EntityManager

@Repository
@RequiredArgsConstructor
class QueryRestaurantRepository(
    private val em: EntityManager
) {
    var restaurant: QRestaurant = QRestaurant.restaurant
    fun findAllRestaurant(restaurantType: String): List<Restaurant>? {
        val queryFactory = JPAQueryFactory(em)


        return queryFactory.selectFrom(restaurant)
            .where(restaurantTypeLike(restaurantType))
            .fetch()
    }

    private fun restaurantTypeLike(restaurantType: String): BooleanExpression? {
        return if (restaurantType.isNotEmpty()) restaurant.restaurantType.eq(restaurantType) else null
    }
}
